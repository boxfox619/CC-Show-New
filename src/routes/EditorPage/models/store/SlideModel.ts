import AssetModel from 'src/core/models/AssetModel';

export default class SlideModel {
    constructor(
        public id: number = 0,
        public name: string = '',
        public thumbnail?: string,
        public assets: AssetModel[] = [],
        public lastAssetId: number = 0,
        public selectedAssetId?: number
    ){}
}