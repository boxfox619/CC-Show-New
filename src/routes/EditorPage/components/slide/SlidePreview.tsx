import * as React from 'react';
import styled from 'styled-components';
import SlideModel from '../../models/Slide';
import IconButton from './IconButton';
import ShareIcon from '../../assets/ic_share_white.png';
import CopyIcon from '../../assets/ic_copy_white.png';
import DeleteIcon from '../../assets/ic_delete_white.png';

const Container = styled.div`
  overflow: hidden;
  height: 140px;
  margin-top: 20px;
  position: relative;
  background: white;
  border-radius: 10px;
  box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.26);
  cursor: pointer;
  ${(props: {active: boolean}) => !props.active && `
    &:after{
        content:'';
        position: absolute;
        left:0;
        top:0;
        width:100%;
        height:100%;
        background-color: rgba(0,0,0,0.2);
        -webkit-animation: fadein 0.4s forwards;
        transition: all 0.5s;
        border-radius: 10px;
    }
    &:hover::after{
        background-color: rgba(0,0,0,0);
    }
  `}
`

const ThumbnailWrapper = styled.div`
    position: absolute;
    overflow: hidden;
    width: 100%;
    height: 100%;
    border-radius: 10px;
`
const Thumbnail = styled.div`
    left: 0px;
    margin-top: -12px;
    width: 100%;
    height: calc( 100% + 12px);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    border-radius: 10px;
    ${(props: {background?: string}) => props.background && `
        background: url(${props.background})
    `}
`
const Controller = styled.div`
    background-color: rgba(0,0,0,0.7);
    bottom: 0;
    width: 100%;
    height: 30%;
    display: table;
    position: absolute;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
`
const Info = styled.div`
    padding: 3px 7px;
    display: table-cell;
    width: 60%;
    vertical-align: middle;
`
const Title = styled.div`
    color: #FFFFFF;
    font-size: 13px;
`
const SubTitle = styled.div`
    color: #9E9E9F;
    font-size: 9px;
`
const Actions = styled.div`
    width: 40%;
    display: table-cell;
    vertical-align: middle;
`

interface Props {
    idx: number,
    slide: SlideModel,
    onClick: () => void,
    active: boolean,
    onShare: () => void,
    onCopy: () => void,
    onDelete: () => void
}

const SlidePreview: React.FC<Props> = (props: Props) => {
    return (
        <Container onClick={props.onClick} active={props.active}>
            <ThumbnailWrapper>
                <Thumbnail background={props.slide.thumbnail} />
            </ThumbnailWrapper>
            <Controller>
                <Info>
                    <Title>{props.slide.name}</Title>
                    <SubTitle>슬라이드{props.idx}</SubTitle>
                </Info>
                <Actions>
                    <IconButton icon={ShareIcon} onClick={props.onShare}/>
                    <IconButton icon={CopyIcon} onClick={props.onCopy}/>
                    <IconButton icon={DeleteIcon} onClick={props.onDelete}/>
                </Actions>
            </Controller>
        </Container>
    )
}

export default SlidePreview;