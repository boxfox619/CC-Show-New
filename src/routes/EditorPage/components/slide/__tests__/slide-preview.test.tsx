import * as React from 'react';
import * as Enzyme from 'enzyme';
import SlidePreview from './../SlidePreview';
import SlideModel from './../../../models/SlideModel';

describe('<SlidePreview />', () => {
  const mockFn = jest.fn();
  it('should match snapshot', () => {
    const mock = new SlideModel(0, '테스트슬라이드', [], 'https://avatars0.githubusercontent.com/u/14067209?s=88&v=4', 0);
    const wrapper = Enzyme.shallow(<SlidePreview idx={1} slide={mock} active={false} onClick={mockFn} onShare={mockFn} onCopy={mockFn} onDelete={mockFn} />);
    expect(wrapper).toMatchSnapshot();
  });
});