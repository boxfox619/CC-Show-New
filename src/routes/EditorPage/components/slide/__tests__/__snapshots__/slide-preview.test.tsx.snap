// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlidePreview /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SlidePreview
    active={false}
    idx={1}
    onClick={[MockFunction]}
    onCopy={[MockFunction]}
    onDelete={[MockFunction]}
    onShare={[MockFunction]}
    slide={
      Slide {
        "assets": "https://avatars0.githubusercontent.com/u/14067209?s=88&v=4",
        "id": 0,
        "lastAssetId": 0,
        "name": "테스트슬라이드",
        "selectedAssetId": undefined,
        "thumbnail": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "children": Array [
        <ForwardRef>
          <ForwardRef
            background={Array []}
          />
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            <ForwardRef>
              테스트슬라이드
            </ForwardRef>
            <ForwardRef>
              슬라이드
              1
            </ForwardRef>
          </ForwardRef>
          <ForwardRef>
            <IconButton
              onClick={[MockFunction]}
            />
            <IconButton
              onClick={[MockFunction]}
            />
            <IconButton
              onClick={[MockFunction]}
            />
          </ForwardRef>
        </ForwardRef>,
      ],
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            background={Array []}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "background": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
    left: 0px;
    margin-top: -12px;
    width: 100%;
    height: calc( 100% + 12px);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    border-radius: 10px;
    ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    position: absolute;
    overflow: hidden;
    width: 100%;
    height: 100%;
    border-radius: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <ForwardRef>
                테스트슬라이드
              </ForwardRef>
              <ForwardRef>
                슬라이드
                1
              </ForwardRef>
            </ForwardRef>,
            <ForwardRef>
              <IconButton
                onClick={[MockFunction]}
              />
              <IconButton
                onClick={[MockFunction]}
              />
              <IconButton
                onClick={[MockFunction]}
              />
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  테스트슬라이드
                </ForwardRef>,
                <ForwardRef>
                  슬라이드
                  1
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "테스트슬라이드",
                },
                "ref": null,
                "rendered": "테스트슬라이드",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": true,
                    "rules": Array [
                      "
    color: #FFFFFF;
    font-size: 13px;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "슬라이드",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "슬라이드",
                  1,
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-gzVnrw",
                    "isStatic": true,
                    "rules": Array [
                      "
    color: #9E9E9F;
    font-size: 9px;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-gzVnrw",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": true,
                "rules": Array [
                  "
    padding: 3px 7px;
    display: table-cell;
    width: 60%;
    vertical-align: middle;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <IconButton
                  onClick={[MockFunction]}
                />,
                <IconButton
                  onClick={[MockFunction]}
                />,
                <IconButton
                  onClick={[MockFunction]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": undefined,
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": undefined,
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": undefined,
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htoDjs",
                "isStatic": true,
                "rules": Array [
                  "
    width: 40%;
    display: table-cell;
    vertical-align: middle;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htoDjs",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
    background-color: rgba(0,0,0,0.7);
    bottom: 0;
    width: 100%;
    height: 30%;
    display: table;
    position: absolute;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  overflow: hidden;
  height: 140px;
  margin-top: 20px;
  position: relative;
  background: white;
  border-radius: 10px;
  box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.26);
  cursor: pointer;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "children": Array [
          <ForwardRef>
            <ForwardRef
              background={Array []}
            />
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              <ForwardRef>
                테스트슬라이드
              </ForwardRef>
              <ForwardRef>
                슬라이드
                1
              </ForwardRef>
            </ForwardRef>
            <ForwardRef>
              <IconButton
                onClick={[MockFunction]}
              />
              <IconButton
                onClick={[MockFunction]}
              />
              <IconButton
                onClick={[MockFunction]}
              />
            </ForwardRef>
          </ForwardRef>,
        ],
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              background={Array []}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "background": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": false,
                "rules": Array [
                  "
    left: 0px;
    margin-top: -12px;
    width: 100%;
    height: calc( 100% + 12px);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    border-radius: 10px;
    ",
                  [Function],
                  "
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    position: absolute;
    overflow: hidden;
    width: 100%;
    height: 100%;
    border-radius: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <ForwardRef>
                  테스트슬라이드
                </ForwardRef>
                <ForwardRef>
                  슬라이드
                  1
                </ForwardRef>
              </ForwardRef>,
              <ForwardRef>
                <IconButton
                  onClick={[MockFunction]}
                />
                <IconButton
                  onClick={[MockFunction]}
                />
                <IconButton
                  onClick={[MockFunction]}
                />
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef>
                    테스트슬라이드
                  </ForwardRef>,
                  <ForwardRef>
                    슬라이드
                    1
                  </ForwardRef>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "테스트슬라이드",
                  },
                  "ref": null,
                  "rendered": "테스트슬라이드",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bZQynM",
                      "isStatic": true,
                      "rules": Array [
                        "
    color: #FFFFFF;
    font-size: 13px;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bZQynM",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "슬라이드",
                      1,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "슬라이드",
                    1,
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-gzVnrw",
                      "isStatic": true,
                      "rules": Array [
                        "
    color: #9E9E9F;
    font-size: 9px;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-gzVnrw",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-EHOje",
                  "isStatic": true,
                  "rules": Array [
                    "
    padding: 3px 7px;
    display: table-cell;
    width: 60%;
    vertical-align: middle;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-EHOje",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <IconButton
                    onClick={[MockFunction]}
                  />,
                  <IconButton
                    onClick={[MockFunction]}
                  />,
                  <IconButton
                    onClick={[MockFunction]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": undefined,
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": undefined,
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": undefined,
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 40%;
    display: table-cell;
    vertical-align: middle;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
    background-color: rgba(0,0,0,0.7);
    bottom: 0;
    width: 100%;
    height: 30%;
    display: table;
    position: absolute;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  overflow: hidden;
  height: 140px;
  margin-top: 20px;
  position: relative;
  background: white;
  border-radius: 10px;
  box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.26);
  cursor: pointer;
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
